### ISSUE TITLE
Line numbers in JSON vs HTML format are different - HTML is correct

### ISSUE THREAD

**Issue by treehousetim** [stoplightio/spectral#1748]
**Line numbers of errors are wrong in JSON output**
Line number is off by +1 when reporting an error in my OpenAPI source document
Character number is off by +1 when reporting an error in my OpenAPI source document.

**To Reproduce**

Using NPM with package.json lines:

```
"_test_html": "spectral lint www/build/openapi.json -o www/test-results.html -v -f html",
"_test_json": "spectral lint www/build/openapi.json -o www/test-results.json -v -f json",
```
with this open api spec

```json
{
  "openapi": "3.0.3",
  "servers": [
    {
      "url": "http://127.0.0.1:8000/v/0.0.1/"
    }
  ],
... [truncated]
```

Creates this output

## JSON
``` json
{
	"code": "oas3-server-trailing-slash",
	"path": [
		"servers",
		"0",
		"url"
	],
	"message": "Server URL should not have a trailing slash.",
	"severity": 1,
	"range": {
		"start": {
			"line": 4,
			"character": 13
		},
		"end": {
			"line": 4,
			"character": 45
		}
	},
	"source": "/Users/timgallagher/Desktop/xampp/wius/api-gateway/swagger/www/build/openapi.json"
},
```

## HTML
```html
<tr style="display:none" class="f-0">
    <td>5:14</td>
    <td class="severity clr-warning">warning</td>
    <td>Server URL should not have a trailing slash.</td>
</tr>
```

**Expected behavior**
I would expect both output formats to be the same.
I would expect JSON to show the same numbers as HTML as the error/warning is actually on line 4 not 5 and at character position 13 not 14.

**Environment (remove any that are not applicable):**
 - Spectral version 5.9.1
 - OS: Mac OSX 11.2.3

**Comment by P0lip** [comment]
Hey!
My apologies for a late answer.
That's on purpose.
In JSON, we use 0-based indices following [LSP](https://microsoft.github.io/language-server-protocol/specifications/specification-3-17/#range).
All other formatters use 1-based indices.

**Comment by P0lip** [comment]
I'll close the issue for now. I hope the explanation makes sense.
We may revisit this in future, but if we ever happen to alter the existing behavior, it'll be clearly noted in changelog / docs.
Shall you have any other concerns or questions, feel free to drop them over here :)
